browser/ephemeral_storage/ephemeral_storage_forget_by_default_browsertest.cc

```diff
@@ -0,0 +1,370 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ephemeral_storage/ephemeral_storage_browsertest.h"
+
+#include "brave/components/brave_shields/browser/brave_shields_util.h"
+#include "chrome/browser/content_settings/cookie_settings_factory.h"
+#include "chrome/browser/content_settings/host_content_settings_map_factory.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/test/base/ui_test_utils.h"
+#include "components/content_settings/core/browser/cookie_settings.h"
+#include "components/content_settings/core/browser/host_content_settings_map.h"
+#include "content/public/test/browser_test.h"
+#include "net/base/features.h"
+
+using content::RenderFrameHost;
+using content::WebContents;
+
+class EphemeralStorageForgetByDefaultBrowserTest
+    : public EphemeralStorageBrowserTest {
+ public:
+  EphemeralStorageForgetByDefaultBrowserTest() {
+    scoped_feature_list_.InitAndEnableFeatureWithParameters(
+        net::features::kBraveForgetFirstPartyStorage,
+        {{"BraveForgetFirstPartyStorageKeepAliveTimeInSeconds", "2"}});
+  }
+  ~EphemeralStorageForgetByDefaultBrowserTest() override = default;
+
+ protected:
+  base::test::ScopedFeatureList scoped_feature_list_;
+};
+
+IN_PROC_BROWSER_TEST_F(EphemeralStorageForgetByDefaultBrowserTest,
```

Comment:
I think it would be nice to also check that waiting for less than the timeout doesn't clear cookies:

- Load exampleA.com and set cookies
- Close tab
- Wait *timeout - 5 seconds*
- Open exampleA.com again
- Check that cookies are still there